tcms/telemetry/static/telemetry/js/testing/test-case-health.js

```diff
@@ -0,0 +1,136 @@
+    if (versionId) {
+        query['case__plan__product_version'] = versionId;
+    }
+
+    const buildId = $('#id_build').val();
+    if (buildId) {
+        query['build_id'] = buildId;
+    }
+
+    const testPlanId = $('#id_test_plan').val();
+    if (testPlanId) {
+        query['case__plan__plan_id'] = testPlanId;
+    }
+
+    const dateBefore = $('#id_before');
+    if (dateBefore.val()) {
+        query['close_date__lte'] = dateBefore.data('DateTimePicker').date().format('YYYY-MM-DD 23:59:59');
+    }
+
+    const dateAfter = $('#id_after');
+    if (dateAfter.val()) {
+        query['close_date__gte'] = dateAfter.data('DateTimePicker').date().format('YYYY-MM-DD 00:00:00');
+    }
+
+    jsonRPC("Testing.test_case_health.grouped", query, data => {
+        drawTable('#table-most-failed', [
+            ...data.all_fail,
+            { text: "Flaky tests" },
+            { header: true },
+            ...data.flaky,
+            { text: "Almost stable" },
+            { header: true },
+            ...data.almost_stable,
+        ]);
+
+        // drawTable('#table-few-failed', data.few_fail);
+        // drawTable('#table-flaky', data.flaky);
+    })
+}
+
+function drawTable(selector, data) {
+    $(`${selector} > tbody`).remove()
+
+    $(selector).DataTable({
+        data: data,
+        columns: [
+            {
+                data: null,
+                render: renderTestCaseColumn
+            },
+            {
+                data: null,
+                render: renderFailedExecutionsColumn
+            },
+            {
+                data: null,
+                render: renderPercentColumn
+            },
+        ],
+        createdRow: (row, data, dataIndex) => {
+            if (data.text) {
+                $(row).css('background-color', '#f5f5f5');
+                $(row).css('font-weight', '600');
+
+                $('td:eq(0)', row).attr('colspan', '3');
+                // hack to hide the other two columns
+                $('td:eq(1)', row).css('display', 'none');
+                $('td:eq(2)', row).css('display', 'none');
+            }
+            if (data.header) {
+                $(row).css('background-color', '#f5f5f5');
+                $(row).css('font-weight', '600');
+
+                $('td:eq(0)', row).text('Test Case');
+                $('td:eq(1)', row).text('Failed executions');
+                $('td:eq(2)', row).text('% of failed executions');
+            }
+        },
+        paging: false,
+        ordering: false,
+        dom: "t"
+    });
+}
+
+function renderTestCaseColumn(data) {
+    if (data.header) {
+        return "";
+    }
+    if (data.text) {
+        return data.text;
+    }
+    return `<a href="/case/${data.case_id}"> TC-${data.case_id} </a>: ${data.case_summary}`;
+}
+
+function renderFailedExecutionsColumn(data) {
+    if (data.header || data.text) {
+        return "";
+    }
+    return `${data.count.fail} of ${data.count.all}`;
```

Comment:
How about a progress bar, similar to the one in the screenshots? Is there anything in Patternfly which can be used quickly ?

Reply:
see my other comment

Reply:
I don't see a comment about progress bar widget so put a link to it.

Reply:
https://github.com/kiwitcms/Kiwi/pull/1096#discussion_r324798497

this one